code/ai/aicode.cpp

```diff
@@ -9613,7 +9613,18 @@ void maybe_update_guard_object(object *hit_objp, object *hitter_objp)
 
 			if ((aip->mode == AIM_GUARD) || (aip->active_goal == AI_ACTIVE_GOAL_DYNAMIC)) {
 				if (aip->guard_objnum == OBJ_INDEX(hit_objp)) {
-					guard_object_was_hit(objp, hitter_objp);
```

Comment:
You should think about whether the fix flag should be checked immediately after `if ((aip->mode == AIM_GUARD) || (aip->active_goal == AI_ACTIVE_GOAL_DYNAMIC)) {`.  As the code is now, you treat ships differently depending on whether they are part of a wing.

Reply:
Good point, I updated the ordering.